### PR TITLE
Remove deprecated methods from PluginCollection

### LINKED ISSUE #22392: Remove deprecated PluginCollection add/remove/addAll/removeAll/clear methods

### PR DESCRIPTION
**Pull request by tresat**
Fixes #22392

### CODE REVIEW

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

File: subprojects/architecture-test/src/changes/accepted-public-api-changes.json [hunk 1]
```diff
@@ -256,6 +256,54 @@
                 "Method has been removed"
             ]
         },
+        {
+            "type": "org.gradle.api.plugins.PluginCollection",
+            "member": "Method org.gradle.api.plugins.PluginCollection.add(java.lang.Object)",
+            "acceptation": "[ADD YOUR CUSTOM REASON HERE]",
```
Comment (big-guy) [inline subprojects/architecture-test/src/changes/accepted-public-api-changes.json:262]: Need reasons here too

**Review by big-guy — CHANGES_REQUESTED** [review]
I think we don't actually want to remove these methods because they're documentation that the PluginCollection class doesn't support those methods.

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
> I think we don't actually want to remove these methods because they're documentation that the PluginCollection class doesn't support those methods.

I wonder if it would be clearer to throw the exceptions via default methods in the interface rather than rely on implementers.  Trying it now.  If it doesn't work, I'll just close this.

**Comment by tresat** [comment]
Doesn't work, the methods from the base domain object class' are chosen rather than the defaults from the interface, so they just have be reimplemented in the plugin collection types anyway.

I'll close this as no further action needed.

**Pull request closed without merging**